### PR TITLE
Scope

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import com.intellij.openapi.util.Condition;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class OpenedFilesScope extends IssueTreeScope {
+
+    public OpenedFilesScope(Project project) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 3. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                l.conditionChanged();
+            }
+        }
+    }
+
+    private static class OpenedFileCondition implements Condition<VirtualFile> {
+        private final List<VirtualFile> openedFiles;
+
+        OpenedFileCondition(List<VirtualFile> openedFiles) {
+            this.openedFiles = openedFiles;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 7. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            VirtualFile[] openFiles = source.getOpenFiles();
+            condition = new OpenedFileCondition(Arrays.asList(openFiles));
+
+            for (ScopeListener l : listeners) {
+                l.conditionChanged();
+            }
+        }
+    }
+
+    private static class OpenedFileCondition implements Condition<VirtualFile> {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 3. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class OpenedFilesScope extends IssueTreeScope {
+
+    public OpenedFilesScope(Project project) {
+        project.getMessageBus().connect(project).subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new EditorChangeListener());
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            condition = new OpenedFileCondition(Arrays.asList(openFiles));
+
+            for (ScopeListener l : listeners) {
+                l.conditionChanged();
+            }
+        }
+    }
+
+    private static class OpenedFileCondition implements Condition<VirtualFile> {
+        private final List<VirtualFile> openedFiles;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+public class OpenedFilesScope extends IssueTreeScope {
+
+    public OpenedFilesScope(Project project) {
+        project.getMessageBus().connect(project).subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new EditorChangeListener());
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Opened files";
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    private class EditorChangeListener extends FileEditorManagerAdapter {
+        @Override
+        public void fileOpened(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+    private static class OpenedFileCondition implements Condition<VirtualFile> {
+        private final List<VirtualFile> openedFiles;
+
+        OpenedFileCondition(List<VirtualFile> openedFiles) {
+            this.openedFiles = openedFiles;
+        }
+
+        @Override public boolean value(VirtualFile virtualFile) {
+            return openedFiles.contains(virtualFile);
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 7. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    public String getDisplayName() {
+        return "Opened files";
+    }
+
+    private class EditorChangeListener extends FileEditorManagerAdapter {
+        @Override
+        public void fileOpened(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
+            refreshCondition(source);
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 7. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    public OpenedFilesScope(Project project) {
+        project.getMessageBus().connect(project).subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new EditorChangeListener());
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Opened files";
+    }
+
+    private class EditorChangeListener extends FileEditorManagerAdapter {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 3. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        @Override
+        public void fileOpened(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
+            refreshCondition(source);
+        }
+
+        @Override
+        public void fileClosed(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
+            refreshCondition(source);
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 7. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        private void refreshCondition(@NotNull FileEditorManager source) {
+            VirtualFile[] openFiles = source.getOpenFiles();
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 7. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+            condition = new OpenedFileCondition(Arrays.asList(openFiles));
+
+            for (ScopeListener l : listeners) {
+                l.conditionChanged();
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/scope/OpenedFilesScope.java]: MAJOR Make this line start at column 9. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 13 issues
* MAJOR 13 major

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarlint/intellij/core/CreateTokenTask.java [hunk 14]
```diff
@@ -56,8 +57,8 @@ public void run(@NotNull ProgressIndicator indicator) {
     try {
       ServerConfiguration.Builder serverConfigBuilder = ServerConfiguration.builder()
         .userAgent("SonarLint IntelliJ " + sonarlint.getVersion())
-        .connectTimeoutMilliseconds(5000)
-        .readTimeoutMilliseconds(5000)
+        .connectTimeoutMilliseconds(SonarLintUtils.CONNETION_TIMEOUT_MS)
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/core/CreateTokenTask.java]: Typo s/CONNETION_TIMEOUT_MS/CONNECTION_TIMEOUT_MS

**Comment by henryju** [comment]
Minor typo + formatting issues but LGTM

**Pull request merged**